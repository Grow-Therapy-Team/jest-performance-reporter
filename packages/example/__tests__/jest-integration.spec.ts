import { readFileSync } from 'fs';

const roundOffNearest100 = (num) => {
  return Math.floor(num / 100) * 100;
};

test('that the json report works fine', () => {
  const report = JSON.parse(readFileSync('performance-report.json').toString());

  expect(
    report.map(({ fullName, status, duration }) => ({
      fullName,
      status,
      duration: roundOffNearest100(duration),
    }))
  ).toMatchSnapshot();
});
