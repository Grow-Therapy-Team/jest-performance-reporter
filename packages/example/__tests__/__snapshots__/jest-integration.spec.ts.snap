// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`that the json report works fine 1`] = `
Array [
  Object {
    "duration": 0,
    "fullName": "Fruits should have a raspberry",
    "status": "passed",
  },
  Object {
    "duration": 500,
    "fullName": "Fruits should have an apple",
    "status": "passed",
  },
  Object {
    "duration": 0,
    "fullName": "should have a raspberry without describe",
    "status": "passed",
  },
  Object {
    "duration": 0,
    "fullName": "Calculator should sum 2+3",
    "status": "passed",
  },
  Object {
    "duration": 3000,
    "fullName": "Calculator should sum 8+0, with timeout",
    "status": "passed",
  },
]
`;
